File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let width = brown / 2;
    let height = 2;
    let area = brown + yellow;
    
    while (true) {
        if (area === (width * height)) break;
        width--;
        height++;
    }
    
    return [width, height];
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    result = []
    for i in range(3, total//2):
        if total % i == 0:
            result.append(total // i)
    
    for y in result:
        x = total // y
        if x >= y and (x-2) * (y-2) == yellow:
            return [x, y]
    
        
            
        